lib/THC/THCReduceApplyUtils.cuh

```diff
@@ -125,7 +237,7 @@ __device__ T reduceBlockWithNThreadLocalReductions(T *smem,
     local = reduceOp(local, next);
   }
 
-  return reduceBlock<T, ReduceOp>(smem, blockDim.x < numVals ? blockDim.x : numVals, local, reduceOp, init);
+  return reduceBlock<T, ReduceOp>(smem, THCCeilDiv(numVals, N), local, reduceOp, init);
```

Comment:
I don't understand the old or new code here.

Isn't `numVals` always less than blockDim.x, because it is the number of threads with N active values? In other words, it has nothing to do with `N`, because all `numVals` threads have N values?